test/mocks/upstream/load_balancer_context.cc

```diff
@@ -1,9 +1,17 @@
 #include "test/mocks/upstream/load_balancer_context.h"
 
+using testing::_;
+using testing::ReturnRef;
+
 namespace Envoy {
 namespace Upstream {
 
-MockLoadBalancerContext::MockLoadBalancerContext() = default;
+MockLoadBalancerContext::MockLoadBalancerContext() {
```

Comment:
Seems unrelated, but I take it it's to keep your other tests simpler?

Reply:
It's for the httpConnPoolForCluster tests I added. If the `LoadBalancerContext` passed in is not null, it ends up choosing a host using the LoadBalancer with a path that requires the context to return something valid here (i.e. it segfaults or trips over asserts if it's not set up correctly). I figured that the vast majority of use-cases for this object would want the returned value here to be 'good', so rather than setting it up for each test, I just made it the default behaviour.

Reply:
sgtm